File: Linked_List/287.find_the_duplicate_number.php
<?php

function findDuplicate($nums)
{
  $slow = $slow2 = $fast = 0;
  while (true) {
    $slow = $nums[$slow];
    $fast = $nums[$nums[$fast]];
    if ($slow === $fast)
      break;
  }
  while (true) {
    $slow = $nums[$slow];
    $slow2 = $nums[$slow2];
    if ($slow === $slow2)
      return $slow;
  }
}

File: Linked_List/287.Find_the_Duplicate_Number.py
class Solution:
    def findDuplicate(self, nums: List[int]) -> int:
        slow, fast = 0,0
        while (True):
            slow, fast = nums[slow], nums[nums[fast]]
            print(slow, fast)
            if slow == fast:
                break
        slow2 = 0
        while(True):
            slow = nums[slow]
            slow2 = nums[slow2]
            if slow == slow2:
                return slow2
